test/e2e/suite-ingress/suite-ingress-resource/stream.go

```diff
@@ -214,4 +223,82 @@ spec:
 		assert.Equal(ginkgo.GinkgoT(), sr[0].ServerPort, int32(9100))
 		assert.Equal(ginkgo.GinkgoT(), sr[0].SNI, "")
 	})
+
+	ginkgo.It("stream tcp proxy with SNI", func() {
+		// create secrets
+		host := "a.test.com"
+		secret := "server-secret"
+		serverCert, serverKey := generateCert(ginkgo.GinkgoT(), []string{host})
+		err := s.NewSecret(secret, serverCert.String(), serverKey.String())
+		assert.Nil(ginkgo.GinkgoT(), err, "create server cert secret error")
+
+		// create ApisixTls resource
+		err = s.NewApisixTls("tls-server", host, secret)
+		assert.Nil(ginkgo.GinkgoT(), err, "create ApisixTls error")
+
+		// check ssl in APISIX
+		assert.Nil(ginkgo.GinkgoT(), s.EnsureNumApisixTlsCreated(1))
+
+		backendSvc, backendSvcPort := s.DefaultHTTPBackend()
+		apisixRoute := fmt.Sprintf(`
+apiVersion: apisix.apache.org/v2
+kind: ApisixRoute
+metadata:
+  name: httpbin-tcp-route
+spec:
+  stream:
+  - name: rule1
+    protocol: TCP
+    match:
+      ingressPort: 9110
+      host: %s
+    backend:
+      serviceName: %s
+      servicePort: %d
+`, host, backendSvc, backendSvcPort[0])
+
+		assert.Nil(ginkgo.GinkgoT(), s.CreateVersionedApisixResource(apisixRoute))
+
+		err = s.EnsureNumApisixStreamRoutesCreated(1)
+		assert.Nil(ginkgo.GinkgoT(), err, "Checking number of routes")
+
+		client := s.NewAPISIXClientWithTLSOverTCP(host)
```

Comment:
I recommend adding the use of other host cases here to ensure that the configured sni is valid.